src/main/java/piggy/Storage.java

```diff
@@ -0,0 +1,126 @@
+package piggy;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import piggy.task.Deadline;
+import piggy.task.Event;
+import piggy.task.Task;
+import piggy.task.TaskWithDate;
+import piggy.task.Todo;
+
+/**
+ * Class that handles storing and retrieving task data from the disk.
+ */
+class Storage {
+    private File file;
+
+    /**
+     * Creates a new Storage object for the given file path.
+     *
+     * @param path The path to the file to read/write to (does not have to exist)
+     */
+    Storage(String path) {
+        file = new File(path);
+    }
+
+    /**
+     * Returns a list of tasks read from the data file.
+     *
+     * @return A list of Tasks.
+     */
+    List<Task> readTasks() {
+        List<Task> tasks = new ArrayList<>();
+        Scanner sc = null;
+        try {
+            sc = new Scanner(file);
+            while (sc.hasNextLine()) {
+                String[] data = sc.nextLine().split(",");
+
+                Task task;
+                boolean done = Boolean.parseBoolean(data[1]);
```

Comment:
does not follow coding standard, better naming would be for boolean "isDone"

Reply:
Agreed, will be fixed.